tests/unit/test_endpoint_provider.py

```diff
@@ -542,29 +555,53 @@ def operation_model_empty_context_params():
     return operation_model
 
 
+US_WEST_2 = "us-west-2"
+US_EAST_1 = "us-east-1"
 BUILTINS_WITH_UNRESOLVED_ACCOUNT_ID = {
-    EndpointResolverBuiltins.AWS_REGION: "us-west-2",
+    EndpointResolverBuiltins.AWS_REGION: US_WEST_2,
     EndpointResolverBuiltins.AWS_ACCOUNT_ID: None,
 }
 BUILTINS_WITH_RESOLVED_ACCOUNT_ID = {
-    EndpointResolverBuiltins.AWS_REGION: "us-west-2",
+    EndpointResolverBuiltins.AWS_REGION: US_WEST_2,
     EndpointResolverBuiltins.AWS_ACCOUNT_ID: "0987654321",
 }
+BUILTINS_WITH_UNRESOLVED_CREDENTIAL_SCOPE = {
+    EndpointResolverBuiltins.AWS_REGION: US_WEST_2,
+    EndpointResolverBuiltins.AWS_CREDENTIAL_SCOPE: None,
+}
+BUILTINS_WITH_RESOLVED_CREDENTIAL_SCOPE = {
+    EndpointResolverBuiltins.AWS_REGION: US_EAST_1,
+    EndpointResolverBuiltins.AWS_CREDENTIAL_SCOPE: US_EAST_1,
+}
 CREDENTIALS = Credentials(
     access_key="access_key",
     secret_key="secret_key",
     token="token",
     account_id="1234567890",
 )
+CREDENTIALS_NO_SCOPE = CREDENTIALS
+CREDENTIALS_WITH_SCOPE = Credentials(
+    access_key="access_key",
+    secret_key="secret_key",
+    token="token",
+    account_id="1234567890",
+    scope=US_WEST_2,
+)
 REQUIRED = "required"
 PREFERRED = "preferred"
 DISABLED = "disabled"
 URL_NO_ACCOUNT_ID = "https://amazonaws.com"
 URL_WITH_ACCOUNT_ID = "https://1234567890.amazonaws.com"
+URL_NO_SCOPE = URL_NO_ACCOUNT_ID
```

Comment:
Is this needed? Should we have a clearer name and define it once? Otherwise, we may accidentally regress the other based on the assumption we're modifying something specific to account id.